src/realm/sort_descriptor.cpp

```diff
@@ -391,6 +403,37 @@ void FilterDescriptor::execute(const Table& table, KeyValues& key_values, const
     key_values = std::move(filtered);
 }
 
+template <typename T, typename U>
```

Comment:
Main comparison function: If we have got the string id for the string, just compare via the interner which will fetch the correct compressed string view.